syscalls_darwin.go

```diff
@@ -127,6 +133,57 @@ func openDev(config Config) (ifce *Interface, err error) {
 	}, nil
 }
 
+// openDevTunTapOSX opens tun / tap device, assuming tuntaposx is installed
+func openDevTunTapOSX(config Config) (ifce *Interface, err error) {
+	var fd int
+	var socketFD int
+
+	if config.DeviceType == TAP && !strings.HasPrefix(config.Name, "tap") {
+		return nil, errors.New("device name does not start with tap when creating a tap device")
+	} else if config.DeviceType == TUN && !strings.HasPrefix(config.Name, "tun") {
+		return nil, errors.New("device name does not start with tun when creating a tun device")
+	} else if config.DeviceType != TAP && config.DeviceType != TUN {
+		return nil, errors.New("unsupported DeviceType")
+	}
```

Comment:
All these `else if` should just be `if`, as they return inside. Also, if I understand correctly `tuntaposx` populates /dev/tap{0-15} and that's all you can use. Perhaps we could be more explicit here in both checking and the error message.